src/screencast.rs

```diff
@@ -0,0 +1,146 @@
+impl ScreenCast {
+    #[dbus_interface(property, name = "version")]
+    fn version(&self) -> u32 {
+        4
+    }
+
+    #[dbus_interface(property)]
+    fn available_cursor_modes(&self) -> u32 {
+        (CursorMode::Hidden | CursorMode::Embedded).bits()
+    }
+
+    #[dbus_interface(property)]
+    fn available_source_types(&self) -> u32 {
+        BitFlags::from_flag(SourceType::Monitor).bits()
+    }
+
+    async fn create_session(
+        &self,
+        request_handle: ObjectPath<'_>,
+        session_handle: ObjectPath<'_>,
+        app_id: String,
+        _options: HashMap<String, Value<'_>>,
+        #[zbus(object_server)] server: &zbus::ObjectServer,
+    ) -> zbus::fdo::Result<(u32, SessionCreateResult)> {
+        tracing::info!(
+            "Start shot: path :{}, appid: {}",
+            request_handle.as_str(),
+            app_id
+        );
+        server
+            .at(
+                request_handle.clone(),
+                RequestInterface {
+                    handle_path: request_handle.clone().into(),
+                },
+            )
+            .await?;
+        let current_session = Session::new(session_handle.clone());
+        append_session(current_session.clone()).await;
+        server.at(session_handle.clone(), current_session).await?;
+        Ok((
+            0,
+            SessionCreateResult {
+                handle_token: session_handle.to_string(),
+            },
+        ))
+    }
+
+    async fn select_sources(
+        &self,
+        _request_handle: ObjectPath<'_>,
+        session_handle: ObjectPath<'_>,
+        _app_id: String,
+        options: SelectSourcesOptions,
+    ) -> zbus::fdo::Result<(u32, HashMap<String, OwnedValue>)> {
+        let mut locked_sessions = SESSIONS.lock().await;
+        let Some(index) = locked_sessions.iter().position(|this_session| this_session.handle_path == session_handle.clone().into()) else {
+            tracing::warn!("No session is created or it is removed");
+            return Ok((2, HashMap::new()));
+        };
+        locked_sessions[index].set_options(options);
+        Ok((0, HashMap::new()))
+    }
+
+    async fn start(
+        &self,
+        _request_handle: ObjectPath<'_>,
+        session_handle: ObjectPath<'_>,
+        _app_id: String,
+        _parent_window: String,
+        _options: HashMap<String, Value<'_>>,
+    ) -> zbus::fdo::Result<(u32, StartReturnValue)> {
+        let locked_sessions = SESSIONS.lock().await;
+        let Some(index) = locked_sessions.iter().position(|this_session| this_session.handle_path == session_handle.clone().into()) else {
+            tracing::warn!("No session is created or it is removed");
+            return Ok((2, StartReturnValue::default()));
+        };
+        let current_session = locked_sessions[index].clone();
+        drop(locked_sessions);
+        let show_cursor = current_session.cursor_mode.show_cursor();
+        println!("{:?}", current_session);
```

Comment:
here will leave a stream, with its fd id..from the doc and read xdg-desktop-portal-kde, and the part of wlr

Reply:
I want there to be a dialog, but this maybe some discussion. what to show